core/stop/residual_norm_reduction.cpp

```diff
@@ -62,20 +62,19 @@ bool ResidualNormReduction<ValueType>::check_impl(
     if (updater.residual_norm_ != nullptr) {
         dense_tau = as<Vector>(updater.residual_norm_);
     } else if (updater.residual_ != nullptr) {
-        u_dense_tau = Vector::create_with_config_of(starting_tau_.get());
-        auto dense_r = as<Vector>(updater.residual_);
-        dense_r->compute_norm2(u_dense_tau.get());
-        dense_tau = u_dense_tau.get();
+        auto *dense_r = as<Vector>(updater.residual_);
```

Comment:
any reason for the extra `*`? It should be the same with or without it

Reply:
Yes, it's just style. I like to make sure/know visually that I am playing with a pointer in those cases. I then feel less wary of passing the object around, etc.